components/eamxx/src/physics/mam/impl/mam4_amicphys.cpp

```diff
@@ -1,1202 +1,1486 @@
+  //          Interstitial aerosols can exist in both cloudy and clear subareas,
+  //          so a grid cell mean needs to be partitioned. Different lognormal
+  //          modes are partitioned differently based on the mode-specific
+  //          mixing ratios. All species in the same mode are partitioned the
+  //          same way, consistent with the internal mixing assumption used in
+  //          MAM.
+  //-----------------------------------------------------------------------------------------
+
+  // jclea, jcldy   : subarea indices
+  // fclea, fcldy   : area fraction [unitless] of the clear and cloudy subareas
+  // qgcm   (ncnst) : grid cell mean, interstitial constituents (unit does not
+  //                  matter)
+  // qqcwgcm(ncnst) : grid cell mean, cloud-borne  constituents (unit
+  //                  does not matter)
+
+  // qgcmx  (ncnst) : grid cell mean, interstitial constituents (unit does not
+  //                  matter)
+  // qsubx(ncnst,maxsubarea): subarea mixing ratios of interst.
+  //                          constituents(unit does not matter as long as they
+  //                          are consistent with the grid cell mean values)
+
+  // Note: qgcm and qqcwgcm are used for calculating the patitioning factors.
+  // qgcmx is the actual grid cell mean that is partitioned into qsubx.
+
+  for(int imode = 0; imode < num_modes; ++imode) {
+    // calculcate partitioning factors
+
+    // grid cell mean of interstitial aerosol mixing ratio of a single mode
+    Real qgcm_intrst = 0;
+
+    // grid cell mean of cloud-borne  aerosol mixing ratio of a single mode
+    Real qgcm_cldbrn = 0;
+
+    // loop thru all species in a mode
+    for(int ispec = 0; ispec < mam4::num_species_mode(imode); ++ispec) {
+      qgcm_intrst = qgcm_intrst + qgcm[lmassptr_amode_gas_pcnst(ispec, imode)];
+      qgcm_cldbrn =
+          qgcm_cldbrn + qqcwgcm[lmassptr_amode_gas_pcnst(ispec, imode)];
     }
+
+    Real factor_clea;  // partitioning factor for clear  subarea [unitless]
+    Real factor_cldy;  // partitioning factor for cloudy subarea [unitless]
+    get_partition_factors(qgcm_intrst, qgcm_cldbrn, fcldy, fclea,  // in
+                          factor_clea, factor_cldy);               // out
+
+    // apply partitioning factors
+    // Here jclea==1 and jcldy==2
+    for(int ispec = 0; ispec < mam4::num_species_mode(imode); ++ispec) {
+      const int icnst     = lmassptr_amode_gas_pcnst(ispec, imode);
+      qsubx[icnst][jclea] = qgcmx[icnst] * factor_clea;
+      qsubx[icnst][jcldy] = qgcmx[icnst] * factor_cldy;
+    }  // ispec
+  }    // imode
+
+}  // set_subarea_qmass_for_intrst_aerosols
+
+//--------------------------------------------------------------------------------
+//--------------------------------------------------------------------------------
+
+KOKKOS_INLINE_FUNCTION
+void set_subarea_gases_and_aerosols(
+    const int &nsubarea, const int &jclea,                           // in
```

Comment:
I love inline comments, but do you think you need to add "//in" for const stuff, and "//out" for non-const ref stuff?

Reply:
I try to add `const` to all inputs, but sometimes a non-const variable can also be an input. This is further complicated by `views` where a `const view_1d` can be output unless it is `const_view_1d`. So, I like explicitly marking them. I can remove it from low-level data types (non-views) if they are marked `const` to reduce the comments.